### PR TITLE
catFR Plots

### PR DESCRIPTION
**Pull request by zduey**
Implemented the IRT and repetition ratio plots. Added subject/experiment/session properties to the base Summary object so that the report generation can more easily distinguish between catFR and FR experiments.

### CODE REVIEW

File: ramutils/reports/generate.py [hunk 1]
```diff
@@ -1,8 +1,10 @@
 from datetime import datetime
 import json
 import os.path as osp
+import numpy as np
 import random
 
+from itertools import compress
```
Comment (mivade) [inline ramutils/reports/generate.py]: Didn't know about this one. Cool.

File: ramutils/reports/generate.py [hunk 2]
```diff
@@ -123,13 +129,53 @@ def _make_combined_summary(self):
...
     def generate(self):
         """Central method to generate any report. The report to run is
         determined by the experiments found in :attr:`session_summary`.
 
         """
         if all(['FR' in exp for exp in self.experiments]):
-            return self.generate_fr1_report()
+            joint = False
+            if any(['catFR' in exp for exp in self.experiments]):
+                joint=True
```
Comment (mivade) [inline ramutils/reports/generate.py]: Insert annoying PEP8 statement here.

File: ramutils/reports/templates/categorical.html [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<div>
+  <h2>Categorical Free Recall Summary</h2>
+  <figure>
+    <div id="catfr-placeholder"></div>
+    <figcaption>
+      <b>(Left)</b> Average inter-response times for transitions within and between categories. <b>(Right)</b>
+      Distribution of repetition ratios across subjects. Dotted line represents the repetition ratios for subject {{ subject }}
```
Comment (mivade) [inline ramutils/reports/templates/categorical.html]: Is listing the subject in the caption really necessary? I know the old reports did this sometimes, but it seems needless to me.

File: ramutils/reports/generate.py (hunk 1 above)
Comment (zduey) [reply ramutils/reports/generate.py]: I didn't either, but luckily the google machine did. I just assume all the weird little helper things I need in python are either in itertools or functools

Comment (mivade) [reply ramutils/reports/generate.py]: Also `collections`.

File: ramutils/reports/generate.py (hunk 2 above)
Comment (zduey) [reply ramutils/reports/generate.py]: What do you mean?

Comment (mivade) [reply ramutils/reports/generate.py]: `joint = True` instead of `joint=True`

Comment (zduey) [reply ramutils/reports/generate.py]: Gotcha, didn't notice the spacing

**Pull request merged**